solutions.js

```diff
@@ -0,0 +1,52 @@
+//***CHALLENGES***
+
+// Write a function called each that takes in an array and a callback function. 
+var each = function (arr, callback){
+// each should iterate through all items in the array
+	for (var i = 0; i < each.length; i++){
+ // call the callback function with each item its index as parameters.		
+		callback(arr[i], i);
+}
+// each should return the original array that was passed in.
+	return(each); 
+}; 
+
+//////////////////////////////////////////////////////////////////////////////////////
+// Write a function called map that takes in an array and a callback function. 
+var map = function (arr, callback){
+	var arrlist = []; 
+// map should iterate through all items in the array, 
+	for (var i = 0; i < map.length; i++){
```

Comment:
Again, logic is awesome. Just one variable name mixed up. We don't have a variable `map`. Look at what our parameters are called.